### DISCUSSION TITLE
Support for Jax / Flax and full differentiable Learning

### DISCUSSION THREAD

**Discussion by rubimat** [Genesis-Embodied-AI/genesis-world#152] [Q&A] [answered]
I want to ask if it is possible to use [Jax](https://github.com/jax-ml/jax) / [Flax](https://github.com/google/flax) etc. libraries instead of PyTorch to close a complete learning pipeline. Jax offers just-in-time compiling and auto-diff for native python functions and it would be nice to use it with the differentiation feature of genesis. Has somebody maybe tried that already?

**Comment by zhouxian** [comment, marked as answer]
Hi, thanks for asking.

For now we prefer to have everything in native taichi and the interface to be in pytorch. If needed, we could probably add interface with jax as well, but for now we are kind of short in terms of manpower :(

**Reply by rubimat** [reply]
I understand... Would be interesting to investigate the difference in performance of Jax (or other) vs PyTorch in the future. But nice project, I will be watching closely :)
